libs/eavmlib/src/network.erl

```diff
@@ -314,13 +295,17 @@ stop() ->
 %%-----------------------------------------------------------------------------
 -spec sta_rssi() -> {ok, Rssi :: db()} | {error, Reason :: term()}.
 sta_rssi() ->
-    Port = get_port(),
-    Ref = make_ref(),
-    Port ! {self(), Ref, rssi},
-    receive
-        {Ref, {error, Reason}} -> {error, Reason};
-        {Ref, {rssi, Rssi}} -> {ok, Rssi};
-        Other -> {error, Other}
+    case whereis(network_port) of
+        undefined ->
+            {error, network_down};
+        Port ->
+            Ref = make_ref(),
```

Comment:
This could also be a monitor on the port.

Reply:
Not sure I understand, this can be called with no network gen_server nor port started, so not sure where a monitor fits in heh (but I'm certainly no expert) - let's make it work first, then make it beautiful;-)

Reply:
The monitor will tell you if the port disappeared since the call to `whereis/1`: you can monitor a gone process or port and you will get a `DOWN` message.

The pattern is the following:
```
Ref = monitor(port, Port),
Port ! {self(), Ref, ...},
Result = receive
   {Ref, ...} ->
   {'DOWN', Ref, port, Port, Reason} -> ...
end,
demonitor(Ref, [flush]),
Result
```

Reply:
Ok, I get it - this is a hypothetical secondary issue where the port disappears unexpectedly after the whereis and stalls the receive - let's call it beyond the scope of this bugfix PR against stable..